taxonomy/utils.py

```diff
@@ -238,3 +243,79 @@ def get_course_jobs(course_key):
+    """
+
+    translation = Translation.objects.filter(source_model_name='Course',
+                                             source_model_field='full_description',
+                                             source_record_identifier=course_key).first()
+    if translation:
+        if translation.source_text != course_description:
+            result = translate_text(course_key, course_description, AUTO, ENGLISH)
+            if not result:
+                return course_description
+            if result['SourceLanguageCode'] == ENGLISH:
+                translation.translated_text = course_description
+            else:
+                translation.translated_text = result['TranslatedText']
+            translation.source_text = course_description
+            translation.source_language = result['SourceLanguageCode']
+            translation.save()
+        return translation.translated_text
+    result = translate_text(course_key, course_description, AUTO, ENGLISH)
+    if not result:
+        return course_description
+    if result['SourceLanguageCode'] == ENGLISH:
+        translated_text = course_description
+    else:
+        translated_text = result['TranslatedText']
+
+    translation = Translation.objects.create(source_model_name='Course',
+                                             source_model_field='full_description',
+                                             source_record_identifier=course_key,
+                                             source_text=course_description,
+                                             translated_text=translated_text,
+                                             translated_text_language=ENGLISH,
+                                             source_language=result['SourceLanguageCode'],
+                                             )
+
+    return translation.translated_text
+
+
+def translate_text(key, text, source_language, target_language):
+    """
+    Translates text into the target language.
+
+    Arguments:
+        key (str): Key or id of the object to uniquely identify.
+        text (str): Text which needs to be translated.
+        source_language (str): Source Language of text if known otherwise provide auto.
+        target_language (str): Desired language in which text needs to be converted.
+
+    Returns:
+        dict: Translated object which contains TranslatedText, SourceLanguageCode and TargetLanguageCode.
+    """
+    translate = boto3.client(service_name=TRANSLATE_SERVICE, region_name=REGION)
+
+    result = None
+    try:
+        result = translate.translate_text(Text=text,
+                                          SourceLanguageCode=source_language,
+                                          TargetLanguageCode=target_language
+                                          )
+    except Exception as ex:
+        message = f'[TAXONOMY] Translate text exception for key: {key} Error: {ex}'
+        LOGGER.error(message)
```

Comment:
can we use  LOGGER.exception(message) here to get fill trace-back details as well.